runtime/compilers/rillv1/template_test.go

```diff
@@ -70,3 +70,15 @@ func TestAnalyze(t *testing.T) {
 		})
 	}
 }
+
+func TestResolve(t *testing.T) {
+	template := "SELECT partner_id FROM domain_partner_mapping WHERE domain = {{ .claims.domain }} and groups IN ({{ .claims.groups | join \",\" }})"
+	resolved, err := ResolveTemplate(template, TemplateData{
+		Claims: map[string]any{
+			"domain": "'rilldata.com'",
+			"groups": []string{"'admin'", "'user'"},
+		},
+	})
+	require.NoError(t, err)
+	require.Equal(t, "SELECT partner_id FROM domain_partner_mapping WHERE domain = 'rilldata.com' and groups IN ('admin','user')", resolved)
+}
```

Comment:
the main reason for doing this is to handle list of values, for example there is no templating syntax to achieve a valid list syntax when user quotes the variable. Here, for the part `groups IN ({{ .claims.groups | join \",\" }})` if user does `groups IN ('{{ .claims.groups | join \",\" }}')` then the resolved expression will be `groups IN ('admin,user')` which is incorrect. Any better way that you can think of to handle lists. 

I was also thinking if the user should just specify `{{ .claims.groups }}` and we should figure out if its a list so add `( )` and join the group values with a `,` but went with this to start with.

Reply:
I think the main thing to decide here is whether we should quote or make the attributes sql compatible for the user or we leave it on the user to create sql compatible syntax and we just replace the template with values as it is. If we leave on user then they might need to understand how sprig templating works and functions they can use in which case the 1st option above looks good and we don't quote single attribute values as well. Otherwise the 3rd option looks good. I did not get how 2nd option is applicable here.